azure-mgmt-servicebus/tests/test_azure_mgmt_servicebus_namespace.py

```diff
@@ -0,0 +1,106 @@
+        self.assertEqual(getnamespaceauthorule.name, authoRule_name, "Authorization rule name not as passed as parameter - get_authorization_rule ")
+        self.assertEqual(len(getnamespaceauthorule.rights), 2, "Access rights mis match as created  - get_authorization_rule ")
+
+        # update the rights of the authorizatiorule
+        getnamespaceauthorule.rights.append('Manage')
+        updatenamespaceauthorule = self.servicebus_client.namespaces.create_or_update_authorization_rule(resource_group_name, namespace_name, authoRule_name, getnamespaceauthorule.rights)
+        self.assertEqual(updatenamespaceauthorule.name, authoRule_name, "Authorization rule name not as passed as parameter for update call - create_or_update_authorization_rule ")
+        self.assertEqual(len(updatenamespaceauthorule.rights), 3, "Access rights mis match as updated  - create_or_update_authorization_rule ")
+
+        #list all the authorization ruels for the given namespace
+        createnamespaceauthorule = list(self.servicebus_client.namespaces.list_authorization_rules(resource_group_name, namespace_name))
+        self.assertEqual(len(createnamespaceauthorule),2, "number of authorization rule mismatch with the created + default = 2 - list_authorization_rules")
+
+        #List keys for the authorization rule
+        listkeysauthorizationrule = self.servicebus_client.namespaces.list_keys(resource_group_name, namespace_name, authoRule_name)
+        self.assertIsNotNone(listkeysauthorizationrule)
+
+        # regenerate Keys for authorizationrule - Primary
+        regenratePrimarykeyauthorizationrule = self.servicebus_client.namespaces.regenerate_keys(resource_group_name, namespace_name,authoRule_name,'PrimaryKey')
+        self.assertNotEqual(listkeysauthorizationrule.primary_key,regenratePrimarykeyauthorizationrule.primary_key)
+
+        # regenerate Keys for authorizationrule - Primary
+        regenrateSecondarykeyauthorizationrule = self.servicebus_client.namespaces.regenerate_keys(resource_group_name,namespace_name,authoRule_name, 'SecondaryKey')
+        self.assertNotEqual(listkeysauthorizationrule.secondary_key, regenrateSecondarykeyauthorizationrule.secondary_key)
+
+        # delete the authorizationrule
+        self.servicebus_client.namespaces.delete_authorization_rule(resource_group_name, namespace_name, authoRule_name)
+
+        # list all the authorization ruels for the given namespace
+        createnamespaceauthorule = list(self.servicebus_client.namespaces.list_authorization_rules(resource_group_name, namespace_name))
+        self.assertEqual(len(createnamespaceauthorule), 1)
+        self.assertEqual(createnamespaceauthorule[0].name, defaultauthorule_name)
+
+        # Delete the create namespace
+        deletenamespace = self.servicebus_client.namespaces.delete(resource_group_name, namespace_name)
```

Comment:
LRO operation needs to be closed explicitly by ".wait()", or you need to add "raw=True" to indicate you don't want to wait. Right now, you leak a thread outside the tests (it's why CI is broken).

Reply:
Resolved it by using raw=True and checking for the state of the namespace before creating entities.